fl4health/clients/basic_client.py

```diff
@@ -635,30 +644,62 @@ def train_by_steps(
 
         return loss_dict, metrics
 
-    def validate(self) -> Tuple[float, Dict[str, Scalar]]:
+    def _val_or_test(
+        self,
+        loader: DataLoader,
+        loss_meter: LossMeter,
+        metric_manager: MetricManager,
+        is_validation: bool = True,
+    ) -> Tuple[float, Dict[str, Scalar]]:
         """
-        Validate the current model on the entire validation dataset.
+        Evaluate the model on the given validation or test dataset.
+
+        Args:
+            loader (DataLoader): The data loader for the dataset (validation or test).
+            loss_meter (LossMeter): The meter to track the losses.
+            metric_manager (MetricManager): The manager to track the metrics.
+            is_validation (bool): Whether this evaluation is for validation or test. Default is True for validation.
```

Comment:
If we create the LoggingMode enum, we can pass that here as well and it will simplify a bit of the code below.

Reply:
I added Enum LooggingMode Class on the top of baseclient